src/hotspot/share/runtime/sharedRuntimeTrans.cpp

```diff
@@ -516,14 +514,16 @@ static double __ieee754_pow(double x, double y) {
   if(lx==0) {
     if(ix==0x7ff00000||ix==0||ix==0x3ff00000){
       z = ax;                   /*x is +-0,+-inf,+-1*/
-      if(hy<0) z = one/z;       /* z = (1/|x|) */
+      if(hy<0) {
+        if (ix == 0) {
```

Comment:
[Just a comment, as the level of change involved seems out of scope for the
issue at hand.  And probably not worth the effort, since x==0.0 is probably
not all that common.  Feel free to completely ignore this.]

I feel like there ought to be a way to restructure this to merge the `ix == 0`
here and the one a few lines earlier.  Something like (completely untested and
quite possibly wrong)
```
  if (ix == 0) {
    z = std::numeric_limits<double>::infinity();
    if (hx < 0 && yisint == 1) {
      z = -z;
    }
    return z;
  } else if (ix == 0x7ff00000 || ix == 0x3ff00000) { // ix==0 moved earlier
    ... ;
    return z;
  }
```